### PR TITLE
feat: remove the need for content-length

### PR DESCRIPTION
**Pull request by maraisr**
From the https://github.com/graphql/graphql-over-http/pull/152 we no longer need content-length things. This PR aims to parse till a boundary, and yield then.

cc @robrichard

### CODE REVIEW

File: src/__test__/PatchResolver.spec.js [hunk 1]
```diff
@@ -65,6 +61,8 @@ describe('PathResolver', function () {
                     boundary,
                 });
 
+                resolver.handleChunk(`\r\n--${boundary}\r\n`);
+
                 resolver.handleChunk(chunk1);
```
Comment (robrichard) [inline src/__test__/PatchResolver.spec.js:66]: Can you add `expect(onResponse).not.toHaveBeenCalled();` above this line to assert the callback is not called after the preamble is sent?

File: src/__test__/PatchResolver.spec.js [hunk 2]
```diff
@@ -164,6 +164,8 @@ describe('PathResolver', function () {
                     boundary,
                 });
 
+                resolver.handleChunk(`\r\n--${boundary}\r\n`);
```
Comment (robrichard) [inline src/__test__/PatchResolver.spec.js:167]: Can you also add a test where the preamble boundary is split into multiple chunks?

Comment (robrichard) [reply src/__test__/PatchResolver.spec.js:167]: Do any of these test a split somewhere in the middle of the boundary?

File: src/PatchResolver.js [hunk 3]
```diff
@@ -3,13 +3,14 @@ import { parseMultipartHttp } from './parseMultipartHttp';
 export function PatchResolver({ onResponse, boundary }) {
     this.boundary = boundary || '-';
     this.onResponse = onResponse;
-    this.processedChunks = 0;
     this.chunkBuffer = '';
+    this.isPreamble = true;
```
Comment (AugustoCalaca) [inline src/PatchResolver.js:7]: can you explain what is and the importance of the `isPreamble`?

Comment (maraisr) [reply src/PatchResolver.js:7]: Everything before the first boundary should be ignored. So this just tracks that we've passed the first boundary. Another way could have been to increment a number, and simply not yield if less than 1.
